scripts/uqfinal.js

```diff
@@ -8,58 +8,92 @@
 var math = require("mathjs");
 
 module.exports = function (robot) {
-	robot.respond(/!?uqfinal ([a-z]{4}[0-9]{4}[a-z]?) (.+)+/i, function (res) {
-		//res.match[2] is the course code
-		//res.match[3] is the space-seperated scores
-		var semester = 6720; // Temporary fix, ID is from rota.eait.uq.edu.au/semesters.json
-		var uqf = robot.http("https://api.uqfinal.com/course/" + semester + "/" + res.match[2].toUpperCase() + ".json")
-				.get() (function(err, resp, body) {
-					if(err) {
-						res.send("Error: " + err);
-                        return;
-					}
-
-                    if(resp.statusCode != 200) {
-						res.send("Invalid Course (Error " + resp.statusCode + ")");
-					    return;
-                    }
-
-					var course = JSON.parse(body).data;
-					var num_of_assessment = course.assessment.length;
-					var scores = res.match[3].split(" ");
-					
-                    if(scores.length != num_of_assessment - 1) {
-						res.send("Invalid number of arguments. Course has " + num_of_assessment + " pieces of assessment, you gave " + scores.length + ".\nKeep in mind you need to leave the last piece of assessment out.");
-					    return;
-                    }
-
-					var total = 0;
-					var invalid = false;
-					for(var i = 0; i < num_of_assessment - 1; i++) {
-						try {
-							var evaluated = math.eval(scores[i]);
-						}catch(err) {
-							res.send("Invalid score: " + scores[i] + ". Reason: Unparsable");
-							invalid = true;
-							break;
-						}
-						if(evaluated > 0 && evaluated < 1) {
-							res.send("_Note: Treating '" + scores[i] + "' (" + evaluated + ") as: " + evaluated*100 + "%_");
-							evaluated *= 100;
-						}
-						if(evaluated < 0 || evaluated > 100) {
-							res.send("Invalid score: " + scores[i] + ". Reason: Score is not within 0 and 100");
-							invalid = true;
-							break;
-						}
-						total += (evaluated/100) * course.assessment[i].weight;
-					}
-
-					if(!invalid) {
-						var needed = 50 - total;
-						var result = Math.ceil(needed/course.assessment[num_of_assessment - 1].weight * 100);
-						res.send("You need to achieve at least " + result + "% on the final exam.\n_Disclaimer: this does not take hurdles into account_\n_Powered by http://uqfinal.com_");
-					}
-				});
-	});
-}
+  robot.respond(/!?uqfinal ([a-z]{4}[0-9]{4}[a-z]?) (.+)+/i, function (res) {
+    var semester;
+    robot.http("https://api.uqfinal.com/semesters")
+      .get()(function (err, resp, body) {
+        if (err) {
+          res.send("Error: " + err);
+          return;
+        }
+
+        // Handle failed HTTP request (check API endpoints at uqfinal.com if consistently failing)
+        if (resp.statusCode != 200) {
+          res.send("Failure fetching current semester (HTTP Error: " + resp.statusCode + ")");
+          return;
+        }
+
+        // Get the current semester uqId (as per UQ Rota)
+        var semester = JSON.parse(body).data.semesters.pop().uqId;
```

Comment:
this will error if `pop()` returns undefined for whatever reason. I'd just put the `uqId` in the `robot.http()` request below :)